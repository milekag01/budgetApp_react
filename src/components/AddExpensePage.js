import React from 'react';
import ExpenseForm from './ExpenseForm';

const AddExpensePage= () => (
    <div>
        <h1>Add Expense Page</h1>
        <ExpenseForm />
    </div>
);

export default AddExpensePage;