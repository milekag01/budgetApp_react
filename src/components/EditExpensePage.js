import React from 'react';

const EditExpensePage= (props) => {
    console.log(props);
    return (
        <div>Edit Expense Page</div>
    );
}

export default EditExpensePage;