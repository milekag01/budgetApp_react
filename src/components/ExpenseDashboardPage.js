import React from 'react';

const ExpenseDashboardPage= () => (
    <div>Expense manager dashboard</div>
);

export default ExpenseDashboardPage;