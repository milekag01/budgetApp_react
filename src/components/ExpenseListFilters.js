import React from 'react';
import {connect} from 'react-redux';
import {setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate} from '../actions/filters';
import { DateRangePicker } from 'react-dates';

export class ExpenseListFilters extends React.Component {
    state = {
        calenderFocused: null
    };

    onDatesChange = ({startDate,endDate}) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };

    onFocusChange = (calenderFocused) => {
        this.setState (() => ({calenderFocused: calenderFocused}));
    };

    onTextChange = (event) => {
        this.props.setTextFilter(event.target.value);
    };

    onSortChange = (event) => {
        if(event.target.value === 'date') {
            this.props.sortByDate();
        } else if(event.target.value === 'amount') {
            this.props.sortByAmount();
        }
    };

    render() {
        return (
            <div>
                <input 
                    type="text" 
                    value={this.props.filters.text}      //this is an example of controlled input where input value is controlled by javascript
                    onChange = {this.onTextChange}
                />

                <select 
                    value={this.props.filters.sortBy}    //this is an example of controlled input where input value is controlled by javascript
                    onChange={this.onSortChange}
                >
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                
                <DateRangePicker 
                    startDate = {this.props.filters.startDate}
                    endDate = {this.props.filters.endDate}
                    onDatesChange = {this.onDatesChange}
                    focusedInput = {this.state.calenderFocused}
                    onFocusChange = {this.onFocusChange}
                    showClearDates = {true}
                    numberOfMonths = {1}
                    isOutsideRange = {() => false}
                />
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate))
});

const connectedExpenseListFilters = connect(mapStateToProps,mapDispatchToProps)(ExpenseListFilters);

export default connectedExpenseListFilters;