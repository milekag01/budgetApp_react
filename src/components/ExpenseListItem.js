import React from 'react';

const ExpenseListItem = ({description,amount,createdAt}) => (
    <div>
        <h1>{description}</h1>
        <p>{amount} - {createdAt}</p>
    </div>
);

export default ExpenseListItem;