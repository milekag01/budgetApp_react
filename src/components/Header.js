import React from 'react';
import { NavLink } from 'react-router-dom';

const Header= () => (
    <div>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Go Home</NavLink>
        <NavLink to="/create" activeClassName="is-active">Add new Expense</NavLink>
        <NavLink to="/edit" activeClassName="is-active">Edit Expense</NavLink>
        <NavLink to="/help" activeClassName="is-active">Get Help</NavLink>
    </div>
);

export default Header;