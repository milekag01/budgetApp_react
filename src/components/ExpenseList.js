import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

const ExpenseList = (props) => (
    <div>
        <h1>Expense List</h1>
        {props.expenses.map((expense) => {
            return <ExpenseListItem key={expense.id} {...expense} />     //passing expense using spread operator so that data can be accessed using destructuring
        })}
    </div>
);

//connect allows us to access the store passed by provider to the components
// return values can of this can be accessed as props in the components above
//now we will break this function to 2 parts : 1. mapStateToProps function and 2. export default
const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses,state.filters)      //We have applied our filters by importing getVisibleItems as selectExpenses and using it 
    }
}

const connectedExpenseList = connect(mapStateToProps)(ExpenseList);

export default connectedExpenseList;

//if we are accessing the value of store then we have to export the connectedComponent