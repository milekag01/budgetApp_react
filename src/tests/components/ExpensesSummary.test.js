import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpenseList summary with 1 expense',() => {
    const wrapper = shallow(<ExpensesSummary expenseCount = {1} expensesTotal = {195} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseList summary with multiple expense',() => {
    const wrapper = shallow(<ExpensesSummary expenseCount = {3} expensesTotal = {54296}/>);
    expect(wrapper).toMatchSnapshot();
});
