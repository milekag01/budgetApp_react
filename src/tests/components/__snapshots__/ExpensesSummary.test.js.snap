// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList summary with 1 expense 1`] = `
<div>
  <h1>
    viewing 
    1
     
    expense
     totalling 
    $1.95
     
  </h1>
</div>
`;

exports[`should render ExpenseList summary with multiple expense 1`] = `
<div>
  <h1>
    viewing 
    3
     
    expenses
     totalling 
    $542.96
     
  </h1>
</div>
`;
